e2e/packages/sync-test/data/encodeTestData.ts

```diff
@@ -10,9 +10,13 @@ export function encodeTestData(testData: Data) {
   return mapObject(testData, (records, table) =>
     records
       ? records.map((record) => ({
-          key: Object.values(record.key).map((key) =>
-            encodeAbiParameters([{ type: "bytes32" }], [toHex(key as any, { size: 32 })])
-          ),
+          key: Object.entries(record.key).map(([keyName, keyValue]) => {
+            const signed = config.tables[table].keySchema[keyName]?.startsWith("int");
+            return encodeAbiParameters(
+              [{ type: "bytes32" }],
+              [signed ? numberToHex(keyValue as any, { size: 32, signed }) : toHex(keyValue as any, { size: 32 })]
+            );
+          }),
```

Comment:
I think our key encoding is now `abi.encode`-compliant. So I think we can just `encodeAbiParameters` with their respective ABI types, don't need to handle values specifically? 

Arguably could add an `encodeKeyTuple` helper to `protocol-parser` to align with `decodeKeyTuple`.

Reply:
good catch!